presto-product-tests/src/main/java/io/prestosql/tests/kafka/TestKafkaWritesSmokeTest.java

```diff
@@ -0,0 +1,271 @@
+    public void testInsertJsonTable()
+    {
+        assertThat(query(format(
+                "INSERT INTO %s.%s.%s VALUES (" +
+                        "'ala ma kota'," +
+                        "9223372036854775807," +
+                        "2147483647," +
+                        "32767," +
+                        "127," +
+                        "1234567890.123456789," +
+                        "true," +
+                        "TIMESTAMP '2018-02-09 13:15:16'," +
+                        "TIMESTAMP '2018-02-09 13:15:17'," +
+                        "TIMESTAMP '2018-02-09 13:15:18'," +
+                        "TIMESTAMP '2018-02-09 13:15:19'," +
+                        "TIMESTAMP '2018-02-09 13:15:20'," +
+                        "DATE '2018-02-11'," +
+                        "DATE '2018-02-13'," +
+                        "TIME '13:15:16'," +
+                        "TIME '13:15:17'," +
+                        "TIME '13:15:18'," +
+                        "TIME '13:15:20'," +
+                        "TIMESTAMP '2018-02-09 13:15:18 UTC'," +
+                        "TIMESTAMP '2018-02-09 13:15:19 UTC'," +
+                        "TIMESTAMP '2018-02-09 13:15:20 UTC'," +
+                        "TIME '13:15:18 +00:00'," +
+                        "TIME '13:15:20 +00:00')",
+                KAFKA_CATALOG,
+                SCHEMA_NAME,
+                ALL_DATATYPES_JSON_TABLE_NAME)))
+                .updatedRowsCountIsEqualTo(1);
+
+        assertThat(query(format(
+                "SELECT * FROM %s.%s.%s",
+                KAFKA_CATALOG,
+                SCHEMA_NAME,
+                ALL_DATATYPES_JSON_TABLE_NAME)))
+                .containsOnly(row(
+                        "ala ma kota",
+                        9223372036854775807L,
+                        2147483647,
+                        32767,
+                        127,
+                        1234567890.123456789,
+                        true,
+                        Timestamp.valueOf(LocalDateTime.of(2018, 2, 9, 13, 15, 16)),
+                        Timestamp.valueOf(LocalDateTime.of(2018, 2, 9, 13, 15, 17)),
+                        Timestamp.valueOf(LocalDateTime.of(2018, 2, 9, 13, 15, 18)),
+                        Timestamp.valueOf(LocalDateTime.of(2018, 2, 9, 13, 15, 19)),
+                        Timestamp.valueOf(LocalDateTime.of(2018, 2, 9, 13, 15, 20)),
+                        Date.valueOf(LocalDate.of(2018, 2, 11)),
+                        Date.valueOf(LocalDate.of(2018, 2, 13)),
+                        Time.valueOf(LocalTime.of(13, 15, 16)),
+                        Time.valueOf(LocalTime.of(13, 15, 17)),
+                        Time.valueOf(LocalTime.of(13, 15, 18)),
+                        Time.valueOf(LocalTime.of(13, 15, 20)),
+                        // different because product test framework converts to java.sql.Timestamp
+                        Timestamp.valueOf(LocalDateTime.of(2018, 2, 9, 19, 0, 18)),
```

Comment:
What is the type of this column?
if it's timestamp *tz*, then we should be using ZonedDateTime and that would require https://github.com/prestosql/presto/pull/307 .......

however, you're testing write path here, not presto JDBC or tempto, so it would be enough (and perhaps actually better) to

```
SELECT CAST(timestamp_tz_column AS varchar) ....
```

and assert on the string representation produced on the Presto side.

Reply:
Nice. That'll also help me confirm if the value differences I was seeing are indeed due to JDBC not supporting ZonedDataTime today.

Reply:
Awesome, it works much better than earlier and doesn't fail even if JVM timezone is changed.

As for https://github.com/prestosql/presto/pull/5838#discussion_r520467509, the cause really is the local time zone of the JVM. The values change according to the test JVM timezone.

Reply:
Was this a TIMESTAMP TZ column?

Reply:
Yes. TIMESTAMP TZ.